### PR TITLE
Vesta/add some tests

### PR DESCRIPTION
**Pull request by vestacic**
First batch of tests for services and validators. 
Currently only Course and Flashcard services are tested, because with other tests it's pretty much the same, so if we lack the coverage, then going to do a little bit of ctrl+c and ctrl+v, same with the validators.

### CODE REVIEW

**Comment by arminasbrazenas** [comment]
Lets complete this PR beforehand https://github.com/vu-code-cartel/studyzen/pull/41

**Review by AkvileR — DISMISSED** [review]

**Review by EvelinaTemnolonskyte — DISMISSED** [review]

File: server/tests/Application.UnitTests/services/CourseServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+using StudyZen.Application.Exceptions;
+using StudyZen.Application.Services;
+using StudyZen.Application.Repositories;
+using StudyZen.Application.Dtos;
+using StudyZen.Application.Validation;
+using StudyZen.Domain.Entities;
+using FluentValidation;
+using System.Linq.Expressions;
+
+namespace StudyZen.tests.Application.UnitTests;
```
Comment (arminasbrazenas) [inline server/tests/Application.UnitTests/services/CourseServiceTests.cs]:
```suggestion
namespace StudyZen.Application.UnitTests.Services;
```

File: server/tests/Application.UnitTests/services/FlashcardServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,251 @@
+using Moq;
+using StudyZen.Application.Services;
+using StudyZen.Application.Repositories;
+using StudyZen.Application.Dtos;
+using StudyZen.Application.Validation;
+using StudyZen.Domain.Entities;
+using FluentValidation;
+using StudyZen.Application.Exceptions;
+
+namespace StudyZen.tests.Application.UnitTests;
```
Comment (arminasbrazenas) [inline server/tests/Application.UnitTests/services/FlashcardServiceTests.cs]:
```suggestion
namespace StudyZen.Application.UnitTests.Services;
```

File: server/tests/Application.UnitTests/validation/validators/Course/CreateCourseRequestValidatorTests.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
+using FluentValidation.TestHelper;
+using StudyZen.Domain.Constraints;
+using StudyZen.Application.Dtos;
+using StudyZen.Application.Validation;
+
```
Comment (arminasbrazenas) [inline server/tests/Application.UnitTests/validation/validators/Course/CreateCourseRequestValidatorTests.cs:7]:
```suggestion
namespace StudyZen.Application.UnitTests.Validation;
```

File: server/tests/Application.UnitTests/validation/validators/Course/UpdateCourseRequestValidatorTests.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
+using FluentValidation.TestHelper;
+using StudyZen.Domain.Constraints;
+using StudyZen.Application.Dtos;
+using StudyZen.Application.Validation;
+
```
Comment (arminasbrazenas) [inline server/tests/Application.UnitTests/validation/validators/Course/UpdateCourseRequestValidatorTests.cs:7]:
```suggestion
namespace StudyZen.Application.UnitTests.Validation;
```

**Review by arminasbrazenas — DISMISSED** [review]

**Review by EvelinaTemnolonskyte — APPROVED** [review]

**Review by arminasbrazenas — APPROVED** [review]

**Pull request merged**